### PR TITLE
Update to OpenSearch 2.10

### PR DESCRIPTION
**Pull request by reta**
### Description
Update to OpenSearch 2.10

### Issues Resolved
N/A

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by reta** [comment]
@nknize just FYI - the things we ask users to absorb as the result of refactorings

**Comment by nknize** [comment]
> @nknize just FYI - the things we ask users to absorb as the result of refactorings 

It's the cost of OpenSearch providing a forward facing API on an Elasticsearch product that wasn't designed to provide a forward facing API. This is going to go on for as long as we don't converge on a forward facing API. We'll get there but folks shouldn't expect it to happen instantaneously.

Maybe we create a meta issue to track the classes being used in downstream projects to make it easier on us properly annotating classes in the core?

**Comment by reta** [comment]
> > @nknize just FYI - the things we ask users to absorb as the result of refactorings
> 
> It's the cost of OpenSearch providing a forward facing API on an Elasticsearch product that wasn't designed to provide a forward facing API. This is going to go on for as long as we don't converge on a forward facing API. We'll get there but folks shouldn't expect it to happen instantaneously.
> 

I understand that for sure, just wanted to have a live example for you :)

> Maybe we create a meta issue to track the classes being used in downstream projects to make it easier on us properly annotating classes in the core?

That could be very difficult to compile I think

File: spring-data-opensearch/src/main/java/org/opensearch/data/client/orhlc/RequestConverters.java [hunk 1]
```diff
@@ -238,9 +238,9 @@ public static Request bulk(BulkRequest bulkRequest) throws IOException {
             if (opType == DocWriteRequest.OpType.INDEX || opType == DocWriteRequest.OpType.CREATE) {
                 IndexRequest indexRequest = (IndexRequest) action;
                 BytesReference indexSource = indexRequest.source();
-                XContentType indexXContentType = indexRequest.getContentType();
+                MediaType indexMediaType = indexRequest.getContentType();
 
-                try (XContentParser parser = XContentHelper.createParser(
+                try (XContentParser parser = org.opensearch.common.xcontent.XContentHelper.createParser(
```
Comment (dblock) [inline spring-data-opensearch/src/main/java/org/opensearch/data/client/orhlc/RequestConverters.java:243]: We don't need the full namespace here, do we?

**Review by dblock — APPROVED** [review]
Small nit?

File: spring-data-opensearch/src/main/java/org/opensearch/data/client/orhlc/RequestConverters.java (hunk 1 above)
Comment (reta) [reply spring-data-opensearch/src/main/java/org/opensearch/data/client/orhlc/RequestConverters.java:243]: Yes we do sadly, we have 2 `XContentHelper`s: class imports `org.opensearch.core.common.xcontent.XContentHelper` but this is another one

**Pull request merged**